logoapp/js/dot_data.js

```diff
@@ -0,0 +1,1028 @@
+					break;
+				}
+			}
+			this.buttonActive.complete = isComplete;
+
+			this.buttonActive.copy = true;
+			this.buttonActive.play = true;
+		}
+	}
+	else {
+		this.currentFrame = clickNum+1;
+		this.arrIsAccess[this.currentFrame-1]=true;
+		var isComplete = true;
+		for(var i=0; i<this.arrIsAccess.length; i++) {
+			if(this.arrIsAccess[i]==false) {
+				isComplete = false;
+				break;
+			}
+		}
+		this.buttonActive.complete = isComplete;
+		this.buttonActive.clear=false;
+		for(var i=0; i<DOT_NUM; i++) {
+			if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+				this.buttonActive.clear=true;
+				break;
+			}
+		}
+	}
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickHelpButton = function() {
+	if(this.buttonActive.help) {
+		dotTutorial.start();
+	}
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickUndoButton = function() {
+	if(this.buttonActive.undo) {
+		if(this.historyPos<this.historyArrDotFrame.length-1) {
+			this.historyPos++;
+			this.arrDotFrame = this.copyArrDotFrame(this.historyArrDotFrame[this.historyPos]);
+		}
+		if(this.historyPos>=this.historyArrDotFrame.length-1) {
+			this.buttonActive.undo=false;
+		}
+		this.buttonActive.clear=false;
+		for(var i=0; i<DOT_NUM; i++) {
+			if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+				this.buttonActive.clear=true;
+				break;
+			}
+		}
+	}
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickClearButton = function() {
+	if(this.buttonActive.clear) {
+		isClear = confirm('編集中のコマを全消去しますか。');
+		if(isClear) {
+			this.buttonActive.clear=false;
+			this.arrDotFrame[this.currentFrame-1] = new DotFrame(LINE_LENGTH, DOT_RADIUS, KIDS_GREEN);;
+			this.addHistory();
+		}
+	}
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickCopyButton = function() {
+	if(this.buttonActive.copy) {
+		if(!this.isCopy) {
+			this.isCopy=true;
+			this.buttonActive = {
+				help: true,
+				undo: false,
+				clear: false,
+				copy: true,
+				play: false,
+				complete: false
+			};
+		}
+		else {
+			this.isCopy=false;
+			for(var i=0; i<DOT_NUM; i++) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+					this.buttonActive.clear=true;
+					break;
+				}
+			}
+			this.buttonActive.copy = true;
+			this.buttonActive.play = true;
+		}
+	}
+	this.drawButton();
+};
+DotAppFramework.prototype.onClickPlayButton = function() {
+	if(this.buttonActive.play) {
+		this.currentFrame = 1;
+		this.playTime = 0;
+		if(this.isPlay) {
+			this.isPlay = !this.isPlay;
+			this.buttonActive.help = true;
+			this.buttonActive.undo=true;
+			if(this.historyPos>=this.historyArrDotFrame.length-1) {
+				this.buttonActive.undo=false;
+			}
+			this.buttonActive.clear=false;
+			for(var i=0; i<DOT_NUM; i++) {
+				if(this.arrDotFrame[this.currentFrame-1].arrDot[i].isDraw) {
+					this.buttonActive.clear=true;
+					break;
+				}
+			}
+			this.buttonActive.copy = true;
+			var isComplete = true;
+			for(var i=0; i<this.arrIsAccess.length; i++) {

```

Comment:
__[jshint]__
*  'i' is already defined.


<!-- Comment by [SideCI](https://sideci.com) -->